### PR TITLE
WIP - Go 1.5

### PR DESCRIPTION
**Pull request by smarterclayton**
Dependent on kubernetes/kubernetes#13058, next rebase will pick it up

### CODE REVIEW

File: hack/verify-govet.sh [hunk 1]
```diff
@@ -18,7 +18,7 @@ cd "${OS_ROOT}"
 mkdir -p _output/govet
 
 FAILURE=false
```
Comment (soltysh) [inline hack/verify-govet.sh]: Line 8:
`if [[ -z $(echo "${GO_VERSION[2]}" | grep -E 'go1.4') ]]; then`
don't forget to update that and `verify-golint.sh` and `verify-gofmt.sh`.

File: hack/common.sh [hunk 2]
```diff
@@ -456,12 +456,12 @@ os::build::ldflags() {
...
-    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.versionFromGit" "${OS_GIT_VERSION}")
-    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.commitFromGit" "${OS_GIT_COMMIT}")
-    ldflags+=(-X "k8s.io/kubernetes/pkg/version.gitCommit" "${KUBE_GIT_COMMIT}")
-    ldflags+=(-X "k8s.io/kubernetes/pkg/version.gitVersion" "${KUBE_GIT_VERSION}")
+    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.majorFromGit=${OS_GIT_MAJOR}")
+    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.minorFromGit=${OS_GIT_MINOR}")
+    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.versionFromGit=${OS_GIT_VERSION}")
+    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.commitFromGit=${OS_GIT_COMMIT}")
+    ldflags+=(-X "k8s.io/kubernetes/pkg/version.gitCommit=${KUBE_GIT_COMMIT}")
+    ldflags+=(-X "k8s.io/kubernetes/pkg/version.gitVersion=${KUBE_GIT_VERSION}")
```
Comment (soltysh) [inline hack/common.sh]: From my tests on s2i, you'll have to conditionally do old version vs this new version.

Comment (soltysh) [inline hack/common.sh]: See https://github.com/openshift/source-to-image/pull/276

Comment (smarterclayton) [inline hack/common.sh]: I didn't see this fail locally...

On Aug 24, 2015, at 6:03 PM, Maciej Szulik <notifications@github.com> wrote:

In hack/common.sh
<https://github.com/openshift/origin/pull/4324#discussion_r37804306>:

> @@ -456,12 +456,12 @@ os::build::ldflags() {
>      os::build::get_version_vars
>
>      declare -a ldflags=()
> -    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.majorFromGit" "${OS_GIT_MAJOR}")
> -    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.minorFromGit" "${OS_GIT_MINOR}")
> -    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.versionFromGit" "${OS_GIT_VERSION}")
> -    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.commitFromGit" "${OS_GIT_COMMIT}")
> -    ldflags+=(-X "k8s.io/kubernetes/pkg/version.gitCommit" "${KUBE_GIT_COMMIT}")
> -    ldflags+=(-X "k8s.io/kubernetes/pkg/version.gitVersion" "${KUBE_GIT_VERSION}")
> +    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.majorFromGit=${OS_GIT_MAJOR}")
> +    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.minorFromGit=${OS_GIT_MINOR}")
> +    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.versionFromGit=${OS_GIT_VERSION}")
> +    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.commitFromGit=${OS_GIT_COMMIT}")
> +    ldflags+=(-X "k8s.io/kubernetes/pkg/version.gitCommit=${KUBE_GIT_COMMIT}")
> +    ldflags+=(-X "k8s.io/kubernetes/pkg/version.gitVersion=${KUBE_GIT_VERSION}")

From my tests on s2i, you'll have to conditionally do old version vs this
new version.

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/4324/files#r37804306>.

Comment (soltysh) [inline hack/common.sh]: Just tried it with your branch, I got http://fpaste.org/258969/. Changing ldflags for go 1.4.2 to the old value worked like a charm.

Comment (soltysh) [inline hack/common.sh]: For reference, this is what upstream is doing, see [here](https://github.com/kubernetes/kubernetes/blob/master/hack/lib/version.sh#L117-L121).

**Comment by ironcladlou** [comment]
@smarterclayton

https://github.com/openshift/origin/blob/master/pkg/cmd/cli/describe/chaindescriber.go#L11

1.5 will disallow internal imports; I fixed it locally by bringing `NodeStack` into the file, but @deads2k might have a nicer way (maybe even not using that struct).

**Comment by mfojtik** [comment]
@smarterclayton go 1.5 landed in F22 :-(

**Comment by mfojtik** [comment]
@smarterclayton F21 got go 1.5 as well, meaning we are broken on all Fedora till this is merged :-)

**Comment by ironcladlou** [comment]
>@smarterclayton F21 got go 1.5 as well, meaning we are broken on all Fedora till this is merged :-)

Speak for yourself... My F21 install is months behind, and I plan to keep it that way until all you early adopters sort out these issues! :grin:

**Comment by mfojtik** [comment]
@ironcladlou problem is that all "new developers"  with F21 are broken as long as they do "yum update".

**Comment by danmcp** [comment]
@smarterclayton Needs rebase.

**Comment by mfojtik** [comment]
@smarterclayton or close :-) I think we merged some portion of this already.

**Pull request closed without merging**